src/pyodide-py/pyodide/simple_web_loop.py

```diff
@@ -0,0 +1,203 @@
+import asyncio
+from asyncio import tasks, futures
+import time
+import contextvars
+
+
+from typing import Awaitable, Callable
+
+
+class SimpleWebLoop(asyncio.AbstractEventLoop):
+    """A custom event loop for use in Pyodide.
+
+    Does no lifecycle management and runs forever (it is just deferring work to the
+    browser event loop which has no lifecycle so why should we have one?)
+
+    run_forever and run_until_complete cannot block like a normal event loop would
+    because we only have one thread so blocking would stall the browser event loop
+    and prevent anything from ever happening.
+
+    We defer all work to the browser event loop using the setTimeout function.
+    To ensure that this event loop doesn't stall out UI and other browser handling,
+    we want to make sure that each task is scheduled on the browser event loop as a
+    task not as a microtask. `setTimeout(callback, 0)` enqueues the callback as a
+    task so it works well for our purposes.
+    """
+
+    def __init__(self, debug: bool = False, interval: int = 10):
+        self._task_factory = None
+        asyncio._set_running_loop(self)
+
+    def get_debug(self):
+        return False
+
+    #
+    # Lifecycle methods: We just ignore all lifecycle management
+    #
+
+    def is_running(self) -> bool:
+        return True
+
+    def is_closed(self) -> bool:
+        return False
+
+    def _check_closed(self):
+        """ Used in create_task. Would raise an error if self.is_closed(), but we are skipping all lifecycle stuff. """
+        pass
+
+    def run_forever(self):
+        """We cannot block like a normal event loop would
```

Comment:
Also need a 1L summary.
```suggestion
        """Run the event loop forever. Does nothing in this implementation.
        
        We cannot block like a normal event loop would
```